lib/tbd/ua.rb

```diff
@@ -26,67 +26,65 @@ module TBD
   #
   # @param model [OpenStudio::Model::Model] a model
   # @param lc [OpenStudio::Model::LayeredConstruction] a layered construction
-  # @param id [String] layered construction identifier
-  # @param heatloss [Double] heat loss from major thermal bridging [W/K]
-  # @param film [Double] target surface film resistance [m2.K/W]
-  # @param ut [Double] target overall Ut for lc [W/m2.K]
+  # @param id [#to_s] layered construction identifier
+  # @param hloss [Numeric] heat loss from major thermal bridging, in W/K
+  # @param film [Numeric] target surface film resistance, in m2•K/W
+  # @param ut [Numeric] target overall Ut for lc, in W/m2•K
   #
-  # @return [Hash] uo: lc Uo [W/m2.K] to meet Ut, m: uprated lc layer
-  # @return [Hash] uo: NilClass, m: NilClass (if invalid input)
+  # @return [Hash] uo: lc Uo [W/m2•K] to meet Ut, m: uprated lc layer
+  # @return [Hash] uo: (nil), m: (nil) if invalid input (see logs)
   def uo(model = nil, lc = nil, id = "", hloss = 0.0, film = 0.0, ut = 0.0)
     mth = "TBD::#{__callee__}"
     res = { uo: nil, m: nil }
     cl1 = OpenStudio::Model::Model
     cl2 = OpenStudio::Model::LayeredConstruction
     cl3 = Numeric
     cl4 = String
-
+    id  = trim(id)
```

Comment:
A new OSlg method that converts _stringable_ Ruby objects into strings, then strips. A LOT of Ruby classes are _stringable_.